master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,101 @@
+
+#### Best practice: implicit default deny policy
+
+Whether you use a network policy or global network policy, we recommend that you create a "plain vanilla" default deny policy that does not include other deny rules. This best practice ensures that unwanted traffic is denied by default. Note that implicit default deny policy always occurs last; if any other policy allows the traffic, then the deny does not come into effect. The deny is executed only after all other policies are evaluated. 
+
+### How to
+
+**Best practice**
+
+Although you can use any of the following policies to create default deny for Kubernetes pods, we recommend using the Calico global network policy. A Calico global network policy applies to both workloads (VMs and containers) and hosts (computers that run the hypervisor for VMs, or container runtime for containers). Using a Calico global network policy supports a conservative security stance for protecting resources. 
+
+- [Create default deny traffic Calico global network policy, non-namespaced](#create-default-deny-traffic-calico-global-network-policy-non-namespaced)
+- [Create default deny traffic Calico network policy, namespaced](#create-default-deny-traffic-calico-network-policy-namespaced)
+- [Create default deny traffic Kubernetes policy, namespaced](#create-default-deny-traffic-Kubernetes-policy-namespaced)
+
+#### Create default deny traffic Calico global network policy, non-namespaced
+
+In the following example, we specify a default deny **GlobalNetworkPolicy** for ingress and egress traffic for all workloads and hosts by using the **selector: all()**. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: default-deny
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+```
+
+#### Create default deny traffic Calico network policy, namespaced  
+
+In the following example, we specify a default deny **NetworkPolicy** that denies all traffic for workloads in the namespace, **engineering**. It is an implicit default deny policy without rules; it will always be in effect, regardless of the ordering scheme that you may specify in normal policies. 
+
+```
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: default-deny
+  namespace: engineering
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress  
+```
+
+#### Create default deny traffic Kubernetes policy, namespaced
+
+The following example is a Kubernetes default deny network policy. It prevents all traffic to/from all pods in the default namespace, and does not explicitly allow any traffic. 
+
+Because the default changes when pods are selected by a network policy, the result is: **deny all ingress and egress traffic**. (Unless the traffic is allowed by another network policy).
```

Comment:
Same sentiment here - we don't say this for the  other examples, but it is equally true in those cases as well. I think that might be confusing and imply that this policy does something the others don't.

Reply:
Fixed.